tracing-core/src/field.rs

```diff
@@ -730,11 +730,46 @@ impl Clone for Field {
     }
 }
 
+// Compares two strings for equality in a constant context.
+const fn str_eq(left: &str, right: &str) -> bool {
```

Comment:
all of this is because the code is executed from a `const fn` context. i couldn't find off-the-shelf functions that do the same, and I also couldn't find a better place to add this check than the constructor of the `FieldSet`.